heimdall/heim_main.c

```diff
@@ -210,12 +210,12 @@ static void heim_fini(Int exitcode)
     if (!current_page_ptr->ordered_next
         || current_page_ptr->ordered_next->start_addr > current_page_ptr->start_addr + clo_page_size)
     {
-      long size = (long)(current_page_ptr->start_addr + clo_page_size - first_page_ptr->start_addr) / 1024;
+      long size = (long)(current_page_ptr->start_addr + clo_page_size - first_page_ptr->start_addr);
 
-      VG_(printf)("  Modified pages: 0x%lx - 0x%lx [size: %ld Kb]\n",
+      VG_(printf)("  Modified area: 0x%lx - 0x%lx size: %ld bytes [%ld.%ld Kb]\n",
                   (long)first_page_ptr->start_addr,
                   (long)(current_page_ptr->start_addr + clo_page_size),
-                  size);
+                  size, size / 1024, ((size % 1024) * 10) / 1024);
```

Comment:
We should put each of the arguments in a new line, the other args are already on their own line.